File: ShopAppSpring/src/main/java/com/duchung/shopappspring/models/Product.java
package com.duchung.shopappspring.models;

import com.duchung.shopappspring.domains.BaseEntity;
import com.fasterxml.jackson.annotation.JsonProperty;
import jakarta.persistence.*;
import lombok.*;

import java.util.List;

@Entity
@Table(name = "products")
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class Product extends BaseEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, length = 350)
    private String name;

    @Column(nullable = false)
    private double price;

    @Column(length = 300)
    private String thumbnail;

    private String description;

    @Column(name = "active")
    @JsonProperty("is_active")
    private int isActive;

    @OneToMany(mappedBy = "product",
               cascade = CascadeType.ALL,
               fetch = FetchType.LAZY)
    private List<ProductImage> productImages;

    @ManyToOne
    @JoinColumn(name = "category_id")
    private Category category;
}

File: shopapp-angular/src/app/user/models/product.ts
import { Category } from './category';
import { ProductImage } from './product.image';

export interface Product {
  id: number;
  name: string;
  description: string;
  url: string;
  created_at: Date;
  updated_at: Date;
  thumbnail: string;
  price: number;
  quantity: number;
  is_active: number;
  category: Category;
  product_images: ProductImage[];
}
